Reqnroll/Infrastructure/DefaultDependencyProvider.cs

```diff
@@ -99,6 +103,24 @@ public virtual void RegisterGlobalContainerDefaults(ObjectContainer container)
             container.RegisterTypeAs<RuntimePluginTestExecutionLifecycleEventEmitter, IRuntimePluginTestExecutionLifecycleEventEmitter>();
 
             container.RegisterTypeAs<TestAssemblyProvider, ITestAssemblyProvider>();
+
+            //Support for publishing Cucumber Messages
+            container.RegisterTypeAs<FileSystem, IFileSystem>();
+            container.RegisterTypeAs<EnvVariableEnableFlagParser, IEnvVariableEnableFlagParser>();
+            container.RegisterTypeAs<FileBasedConfigurationResolver, ICucumberMessagesConfigurationResolver>("fileBasedResolver");
+            container.RegisterTypeAs<EnvironmentConfigurationResolver, ICucumberMessagesConfigurationResolver>("environmentBasedResolver");
+            container.RegisterFactoryAs<IEnumerable<ICucumberMessagesConfigurationResolver>>(() =>
```

Comment:
I think this might not be needed. BoDi can return all the named registrations if you resolve an `IDictionary<string, IInterface>`. See https://github.com/reqnroll/Reqnroll/blob/7a9293df260385ec1c1b32f3e25eddea8b9258ec/Tests/Reqnroll.RuntimeTests/BoDi/NamedRegistrationTests.cs?plain=1#L89C1-L89C86

Reply:
If we want to make this extensible, I think for this case I would go for the `Priority` property on the `ICucumberMessagesConfigurationResolver` interface to ensure the order.

If not (or for now), the hard-coded solution would also work, but in that case I think I would _not_ use the named registration, but I would make two separate interface (that derive from `ICucumberMessagesConfigurationResolver`) for the two resolvers and I would depend on them individually. That way it seems to be clear that right now we support this two and only this two resolvers.

Reply:
See commit [be7cb11](https://github.com/reqnroll/Reqnroll/pull/233/commits/be7cb113e84a39ca7a995acc0a1357ca03abc9cf)

Reply:
I see. I think it is not possible to extend the resolver system without knowing the order, so the extensibility point for the "other" resolvers is not really usable this way. For now I would vote for making the file based resolver also explicit, like you did with `ICucumberMessagesEnvironmentOverrideConfigurationResolver` and defer the extensibility to the time when we globally rethink the configuration infrastructure and the environment variable handling.